src/data_distribution/naive_distribution.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * @file
+ * @brief This file implements the naive data distribution
+ * @copyright Eta Scale AB. Licensed under the Eta Scale Open Source License. See the LICENSE file for details.
+ */
+
+#ifndef argo_naive_distribution_hpp
+#define argo_naive_distribution_hpp argo_naive_distribution_hpp
+
+namespace argo {
+	namespace data_distribution {
+		/* forward declaration */
+		template<int instance> class base_distribution;
+		/* forward declaration */
+		template<typename T, class Dist = base_distribution<0>> class global_ptr;
```

Comment:
As mentioned before, there should be a better place for these rather than in a specific distribution implementation.

Reply:
Changes were made.
Both of the forward declarations are removed:
- The new file hierarchy removes the `base_distribution` forward declaration.
- The `global_ptr` forward declaration is removed by passing `base_distribution<0>` directly to the `global_ptr` class definition, as per the original format.

Reply:
Same as above here, unless @davidklaftenegger has any comments I will say that this is okay for now.